pydicom/dataset.py

```diff
@@ -2120,6 +2186,16 @@ def __setattr__(self, name: str, value: Any) -> None:
         value
             The value for the attribute to be added/changed.
         """
+        # Save time for common Dataset attributes that are not DICOM keywords
+        # This check is fast if `name` is a DICOM keyword (first chr is upper)
+        # The startswith is needed for `is_implicit_VR`
```

Comment:
That would help with the `islower` thing, however, having `startswith` is probs faster for the common `is_...` case as it only needs to check the first few characters rather than all of them (for those that are indeed not a DICOM keyword).  We could go really fast and just check if the first character is lower case, but then we would lose warnings/errors for camel case like `ds.patientName`.